tests/test_train.py

```diff
@@ -358,3 +367,347 @@ def test_train_nlu_autoconfig(
     mocked_get_configuration.assert_called_once()
     _, args, _ = mocked_get_configuration.mock_calls[0]
     assert args[1] == autoconfig.TrainingType.NLU
+
+
+def test_model_finetuning(
+    tmp_path: Path,
+    monkeypatch: MonkeyPatch,
+    default_domain_path: Text,
+    default_stories_file: Text,
+    default_stack_config: Text,
+    default_nlu_data: Text,
+    trained_rasa_model: Text,
+):
+    mocked_nlu_training = Mock()
+
+    async def mocked_train_nlu(*args: Any, **kwargs: Any) -> Text:
```

Comment:
This is repeated in all the tests so could be extracted into a helper. I did something similar (based on this) in https://github.com/RasaHQ/rasa/pull/7382